instrumentation/aws-java-sdk-dynamodb-2.15.34/src/main/java/software/amazon/awssdk/services/dynamodb/DefaultDynamoDbAsyncClient_Instrumentation.java

```diff
@@ -0,0 +1,139 @@
+package software.amazon.awssdk.services.dynamodb;
+
+import com.newrelic.api.agent.NewRelic;
+import com.newrelic.api.agent.Trace;
+import com.newrelic.api.agent.weaver.MatchType;
+import com.newrelic.api.agent.weaver.Weave;
+import com.newrelic.api.agent.weaver.Weaver;
+import com.nr.instrumentation.dynamodb_v2.DynamoDBMetricUtil;
+import software.amazon.awssdk.core.client.config.SdkClientConfiguration;
+import software.amazon.awssdk.core.client.config.SdkClientOption;
+import software.amazon.awssdk.services.dynamodb.model.*;
+
+import java.net.URI;
+import java.util.concurrent.CompletableFuture;
+
+@Weave(originalName = "software.amazon.awssdk.services.dynamodb.DefaultDynamoDbAsyncClient", type = MatchType.ExactClass)
+final class DefaultDynamoDbAsyncClient_Instrumentation {
+
+    private final SdkClientConfiguration clientConfiguration = Weaver.callOriginal();
+
+    @Trace
+    public CompletableFuture<ScanResponse> scan(ScanRequest scanRequest) {
+        DynamoDBMetricUtil.metrics(NewRelic.getAgent().getTracedMethod(), "scan", scanRequest.tableName(), getEndpoint());
```

Comment:
is the query time being properly accounted for here?
Does Dynamo uses the same thread that called and returns a calculated future?

Reply:
I honestly don't know.  @XiXiaPdx would need to help clear that up.